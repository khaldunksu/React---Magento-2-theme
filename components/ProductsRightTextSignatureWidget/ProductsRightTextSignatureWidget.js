import React from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import ReactHtmlParser from "react-html-parser";
import "./ProductsRightTextSignatureWidget.scss";
import HomeSection from "../HomeSection/HomeSection";
import ProductCard from "../ProductCard/ProductCard";


export function ProductsRightTextSignatureWidget(props) {
    const { data } = props;

    return (
        <HomeSection classNames="js-hook__right-text-signature right-text-signature-widget">
            {data.products.length ?
                <div className="right-text-signature-widget__products js-hook__right-text-signature-products">
                    {data.products.map((item, i)=>(
                    <ProductCard className="right-text-signature-widget__products-item" product={item} key={"right-text-widget-product"+i}/>
                    ))}
                </div>
                : null}
                <div className="right-text-signature-widget__content">
                    <div className="js-hook__right-text-signature-text">{ReactHtmlParser(data.text)}</div>
                    <img className="js-hook__right-text-signature-image" src={data.image} alt=""/>
                </div>
        </HomeSection>
    );
}

ProductsRightTextSignatureWidget.propTypes = {
    data: PropTypes.object
};
ProductsRightTextSignatureWidget.defaultProps = {
    data: {}
};

export default connect(state => state)(ProductsRightTextSignatureWidget);
