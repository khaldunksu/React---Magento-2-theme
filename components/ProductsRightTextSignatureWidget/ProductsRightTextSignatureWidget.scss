.right-text-signature-widget{
  background: url("/static/img/section2-back.png");
  background-repeat: no-repeat;
  background-size: cover;

  &__products{
    display: flex;
    width: 50%;
    flex-wrap: wrap;

    &-item{
      margin: 10px;
    }
  }
}