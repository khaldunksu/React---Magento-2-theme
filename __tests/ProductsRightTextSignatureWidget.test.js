import { shallow } from "enzyme";
import React from "react";
import Router from "next/router";
import toJson from "enzyme-to-json";
import { ProductsRightTextSignatureWidget } from "../components/ProductsRightTextSignatureWidget/ProductsRightTextSignatureWidget";

const mockedRouter = { push: () => {} };
Router.router = mockedRouter;
const data = {
    image: "some-image-url",
    name: "Test Section",
    products: [
        {
            name: "Test",
            price: "0"
        }

    ],
    text: "Test text"
};

describe("Component ProductsRightTextSignatureWidget matches snapshot", () => {
    it("renders page layout correctly", () => {
        const wrapper = shallow(<ProductsRightTextSignatureWidget data={data}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

describe("Component ProductsRightTextSignatureWidget", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<ProductsRightTextSignatureWidget data={data}/>);
    });

    it("renders the dumb component", () => {
        expect(wrapper.length).toEqual(1);
        expect(wrapper.find(".js-hook__right-text-signature-products").length).toBe(1);
        expect(wrapper.find(".js-hook__right-text-signature-text").length).toBe(1);
        expect(wrapper.find(".js-hook__right-text-signature-image").length).toBe(1);
    });

});