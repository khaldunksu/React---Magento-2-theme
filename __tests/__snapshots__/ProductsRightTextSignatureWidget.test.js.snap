// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ProductsRightTextSignatureWidget matches snapshot renders page layout correctly 1`] = `
<HomeSection
  classNames="js-hook__right-text-signature right-text-signature-widget"
  contentClass=""
>
  <div
    className="right-text-signature-widget__products js-hook__right-text-signature-products"
  >
    <ProductCard
      className="right-text-signature-widget__products-item"
      key="right-text-widget-product0"
      product={
        Object {
          "name": "Test",
          "price": "0",
        }
      }
    />
  </div>
  <div
    className="right-text-signature-widget__content"
  >
    <div
      className="js-hook__right-text-signature-text"
    >
      Test text
    </div>
    <img
      alt=""
      className="js-hook__right-text-signature-image"
      src="some-image-url"
    />
  </div>
</HomeSection>
`;
